core/src/main/kotlin/org/evomaster/core/search/gene/ObjectGene.kt

```diff
@@ -187,25 +187,36 @@ class ObjectGene(
         return additionalFields!!.any { it.first.value == fieldToAdd.first.value}
     }
 
-    override fun copyValueFrom(other: Gene) {
+    override fun copyValueFrom(other: Gene): Boolean {
         if (other !is ObjectGene) {
             throw IllegalArgumentException("Invalid gene type ${other.javaClass}")
         }
 
         if (other.isFixed != isFixed)
             throw IllegalArgumentException("cannot copy value for ObjectGene if their isFixed is different")
 
+        val current = copy()
+
+        var ok = true
+
         for (i in fixedFields.indices) {
-            this.fixedFields[i].copyValueFrom(other.fixedFields[i])
+            ok = ok && this.fixedFields[i].copyValueFrom(other.fixedFields[i])
+        }
+
+        if (!ok || !isLocallyValid()){
+            Lazy.assert { copyValueFrom(current) }
```

Comment:
an modifying state operation inside an assertion? that would not apply at runtime outside the running of tests (which have `-ea` in maven and by default in IntelliJ).
or did u rather meant something like `val x = copyValueFrom(current);  assert(x);`?